### ISSUE TITLE
Removing the automatic use of spaces as word separators

### ISSUE THREAD

**Issue by thetruejacob** [githubharald/CTCWordBeamSearch#60]
Hi, I'm continuing this topic from [here](https://github.com/githubharald/SimpleHTR/issues/122).

TL;DR I would like to use WBS for writing systems that do not use spaces as word separators. 

Could you point me in the right direction - where in the code does a space get inserted before moving on to predicting the next word? I was simply assuming if this piece of code was removed, I would get what I want - a long string of words, and spaces are simply treated as another character. I actually believe I get this behavior with normal beamsearch/bestpath decoder - everything looks nice (but of course less accurate), while the WBS decoder is more accurate but introduces spaces unnecessarily.

Is there a way to use spaces **only** as a character, indistinguishable from any other character? Why do they keep getting introduced when moving onto the next beam?

This would definitely be helpful for my use case for the Thai language, and I'm sure many other users who want to adapt it to [certain languages](https://en.wikipedia.org/wiki/Category:Writing_systems_without_word_boundaries) (Thai, Lao, Tibetan, Burmese, Mongolian etc) would also highly appreciate it.

Thank you again for your continued maintenance on this project.

**Comment by githubharald** [comment]
Hi, it's quite a while since I worked on that codebase.
As I pointed out in the prev. thread, I think there will be some changes needed both in the language model and the beam classes. But I did not see an "easy fix" back than, so it really means going through the code with some debugger, looking what's going on, and then adapting the code. 
Basically, it should be possible, because it just means allowing new words as soon as the previous word in finished. But much of the code relies on the space character, so it might be some work to change this.
